### ISSUE TITLE
[CI] DeprecationHttpIT testDeprecationRouteThrottling failing

### ISSUE THREAD

**Issue by iverase** [elastic/elasticsearch#100849]
**Build scan:**
https://gradle-enterprise.elastic.co/s/c6eimwkt2me3e/tests/:x-pack:plugin:deprecation:qa:rest:javaRestTest/org.elasticsearch.xpack.deprecation.DeprecationHttpIT/testDeprecationRouteThrottling
**Reproduction line:**
```
./gradlew ':x-pack:plugin:deprecation:qa:rest:javaRestTest' --tests "org.elasticsearch.xpack.deprecation.DeprecationHttpIT.testDeprecationRouteThrottling" -Dtests.seed=D96B34BFA26D46B5 -Dtests.locale=ro -Dtests.timezone=Indian/Mauritius -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.deprecation.DeprecationHttpIT&tests.test=testDeprecationRouteThrottling
**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: iterable with items [(map containing ["event.code"->"deprecated_route_POST_/_test_cluster/deprecated_settings"] and map containing ["message"->"[/_test_cluster/deprecated_settings] exists for deprecated tests"]), (map containing ["event.code"->"deprecated_route_GET_/_test_cluster/deprecated_settings"] and map containing ["message"->"[/_test_cluster/deprecated_settings] exists for deprecated tests"]), (map containing ["event.code"->"deprecated_settings"] and map containing ["message"->"[deprecated_settings] usage is deprecated. use [settings] instead"])] in any order
     but: not matched: <{elasticsearch.cluster.name=javaRestTest, service.name=ES_ECS, elasticsearch.event.category=settings, data_stream.namespace=default, message=[test.setting.deprecated.true2] setting was deprecated in Elasticsearch and will be removed in a future release., data_stream.type=logs, log.logger=org.elasticsearch.deprecation.common.settings.Settings, @timestamp=2023-10-13T16:58:09.356Z, event.code=test.setting.deprecated.true2, elasticsearch.cluster.uuid=zlWPayGgQ7KwQ4SYuGYK2w, elasticsearch.node.name=javaRestTest-0, ecs.version=1.2.0, data_stream.dataset=deprecation.elasticsearch, log.level=WARN, event.dataset=deprecation.elasticsearch, process.thread.name=elasticsearch[javaRestTest-0][masterService#updateTask][T#1], elasticsearch.node.id=VIZSuuLaSmmAuU6NYNow8g}>

  at __randomizedtesting.SeedInfo.seed([D96B34BFA26D46B5:C9B2C539A3251A62]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.deprecation.DeprecationHttpIT.lambda$testDeprecationRouteThrottling$7(DeprecationHttpIT.java:319)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1197)
  at org.elasticsearch.xpack.deprecation.DeprecationHttpIT.testDeprecationRouteThrottling(DeprecationHttpIT.java:314)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

**Comment by thecoop** [comment]
Looks like some leakage between tests

**Comment by thecoop** [comment]
Hopefully fixed by #101273
